cmd/crictl/container.go

```diff
@@ -808,9 +808,20 @@ func CreateContainer(
 		}
 
 		// Try to pull the image before container creation
-		ann := config.GetImage().GetAnnotations()
-		if _, err := PullImageWithSandbox(iClient, image, auth, podConfig, ann, opts.pullOptions.timeout); err != nil {
-			return "", err
+		images := []string{image}
+
+		// Add possible OCI volume mounts
+		for _, m := range config.Mounts {
+			if m.Image != nil && m.Image.Image != "" {
+				images = append(images, m.Image.Image)
+			}
+		}
+
+		for _, image := range images {
+			logrus.Infof("Pulling image: %s", image)
```

Comment:
can this message be a bit clearer that it's being pulled for a mount? also: maybe we should only do this if the pull-on-create flag is set?

Reply:
I changed the log messages to reflect which image / OCI object we'd like to pull on which purpose.

I'm not sure to which flag you're referring. Right now it's only enabled when `--with-pull` is specified on `create` or `run`.